client/src/Try/Component.purs

```diff
@@ -0,0 +1,542 @@
+        case eitherId of
+          Left err -> log err
+          Right id -> do
+            putContentSource $ HaveGist id
+            liftEffect $ pushRoute $ "/?" <> gistQP <> "=" <> (show id)
+
+      -- Create editor annotations and markers for errors or warnings
+      annotateAndMark :: forall r. AnnotationType -> Array (WarningOrError r) -> _
+      annotateAndMark type_ arr = do
+        ses <- HK.get sessionIdx
+        case ses of
+          Nothing -> do
+            log "Error: Session unset"
+          Just s ->
+            liftEffect do
+              -- annotations
+              let
+                annotations = mapMaybe (toAnnotation type_) arr
+              Session.setAnnotations annotations s
+              -- markers
+              maybeRanges <- traverse mkMarkerRange arr
+              let
+                (ranges :: Array Range) = catMaybes maybeRanges
+              -- The shown type of "error" or "warning" should match the css class
+              -- found in ace.css.
+              -- `true` to show in "front"
+              traverse_ (\r -> Session.addMarker r (show type_) "text" true s) ranges
+
+      -- Clear annotations and marks from editor
+      removeAnnotationsAndMarks = do
+        ses <- HK.get sessionIdx
+        case ses of
+          Nothing -> log "Error: Session unset"
+          Just s ->
+            liftEffect do
+              -- Remove annotations
+              Session.clearAnnotations s
+              -- A `removeMarkers` function would be nice...
+              -- Get all inFront(=true) markers
+              markers <- MyAce.getMarkers true s
+              (markerIds :: Array Int) <- traverse Marker.getId markers
+              -- Remove markers
+              traverse_ (\m -> Session.removeMarker m s) markerIds
+
+      doCompile = do
+        -- Needed now that this is called from debouncer.
+        -- Stale otherwise.
+        currentContent <- HK.get contentIdx
+        -- Clear annotations and marks from editor
+        removeAnnotationsAndMarks
+        putOutput Loading
+        (res :: Either String CompileResult) <- liftAff $ compile currentContent
+        case res of
+          Right (CompileSuccess { js, warnings }) -> do
+            -- warnings is maybe array. Why not just let this be an array?
+            -- Create annotations and markers for warnings
+            annotateAndMark AnnotateWarning $ fromMaybe [] warnings
+            -- Get JS source files for all included modules
+            obj <- liftAff $ runLoader loader $ JS js
+            -- Compress source file contents.
+            -- This avoids non-trivial HTML + JSON string escaping
+            let
+              objcomp = map (unwrap >>> compressToEncodedURIComponent) obj
+            -- Convert object to JSON string
+            let
+              jsonStr = stringify $ encodeJson objcomp
+            -- Pass JS files to execute in iframe
+            putOutput $ App jsonStr $ JS js
```

Comment:
The JS files are compressed with lz-string because I couldn't figure out how to escape the contents correctly to pass to the iframe. It would be nice to escape these a bit more efficiently, although I don't think this compression/decompression step is a major performance issue.